binaries/cuprated/src/config.rs

```diff
@@ -262,6 +264,22 @@ impl Config {
     }
 }
 
+impl fmt::Display for Config {
+    fn fmt(&self, config: &mut fmt::Formatter<'_>) -> fmt::Result {
+        info!("========== CONFIGURATION ==========");
+        info!("Network: {}", self.network);
+        info!("Fast Sync: {}", self.fast_sync);
+        info!("Tracing Config: {:?}", self.tracing);
+        info!("Tokio Config: {:?}", self.tokio);
+        info!("Rayon Config: {:?}", self.rayon);
+        info!("P2P Config: {:?}", self.p2p);
+        info!("Storage Config: {:?}", self.storage);
+        info!("File System Config: {:?}", self.fs);
+        info!("===================================");
+        Ok(())
+    }
```

Comment:
here you are writing to the log output, whereas you should be writing to the `Formatter` passed in:

```suggestion
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        writeln!(f, "========== CONFIGURATION ==========")?;
        writeln!(f, "Network: {}", self.network)?;
        writeln!(f, "Fast Sync: {}", self.fast_sync)?;
        writeln!(f, "Tracing Config: {:?}", self.tracing)?;
        writeln!(f, "Tokio Config: {:?}", self.tokio)?;
        writeln!(f, "Rayon Config: {:?}", self.rayon)?;
        writeln!(f, "P2P Config: {:?}", self.p2p)?;
        writeln!(f, "Storage Config: {:?}", self.storage)?;
        writeln!(f, "File System Config: {:?}", self.fs)?;
        writeln!(f, "===================================")?;
        Ok(())
    }
```